go/mcap/writer.go

```diff
@@ -589,6 +588,7 @@ func (w *Writer) Close() error {
 	}
 
 	// summary section
+	w.w.Reset() // reset CRC to begin computing summaryCrc
```

Comment:
we should probably rename this to ResetCRC

Reply:
also, preexisting but it's confusing that WriteSizer knows anything about CRCs I guess. Maybe this is a WriteCRCSizer, or we can compose somehow.. dunno, maybe for another time.